lib/callback_model.js

```diff
@@ -199,13 +243,33 @@ Channel.prototype.cancel = function(consumerTag, cb0) {
     function(err, ok) {
       if (err === null) {
         self.unregisterConsumer(consumerTag);
+        //TODO: If autorecovery is enabled
+        this.remove_consumer(consumerTag);
         cb(null, ok.fields);
       }
       else cb(err);
     });
   return this;
 };
 
+Channel.prototype.record_consumer = function(tag, queue, callback, options){
+  this.ensureConsumersBooked();
+  this.book.consumers[tag] = {queue: queue, callback: callback, options: options};
+};
+
+Channel.prototype.remove_consumer = function(tag){
+  delete this.book.consumers[tag];
+};
+
+Channel.prototype.ensureConsumersBooked = function(){
+  if(this.book == undefined) {
```

Comment:
prefer === 
See https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons